balloc/balloc.c

```diff
@@ -2283,9 +2388,9 @@ static int balloc_try_best_found(struct balloc_allocation_context *bac,
 
 		balloc_debug_dump_extent(__func__, &bac->bac_final);
 		M0_ASSERT(is_extent_free(grp, &bac->bac_final, alloc_flag,
-					 &cur));
+					 &cur, &next));
```

Comment:
M0_ASSERT(), and other ASSERT macro, should have no side-effect.
The M0_ASSERT() did not change things in grp, or bac, but it sill changes the cur, and next.
So, it's better to save the result from is_extent_free() and then assert the result.